miner: fix miner application of DAO state

meowsbits opened the pull request:
This caused ETH-supporting chains to incorrectly apply
the DAO state manipulation during mining.

soc1c commented:
FYI https://github.com/ethereumclassic/ECIPs/issues/284

soc1c commented on params/config_mordor.go:
```diff
@@ -57,12 +57,16 @@ var (
 		EIP2028FBlock: big.NewInt(778507),
-		EIP2200FBlock: big.NewInt(778507), // RePetersburg (== re-1283)
+		// ECIP-1078, ECIP-1079
+		//EIP2200FBlock: big.NewInt(778507), // RePetersburg (== re-1283)
```
this won't work for mordor anymore.

soc1c commented:
proposed block numbers for phoenix/ecip-1078
- `976_231` on Mordor Classic PoW-testnet (March 4th, 2020)
- `2_208_203` on Kotti Classic PoA-testnet (March 11th, 2020)
- `10_500_839` on Ethereum Classic PoW-mainnet (March 25th, 2020)

ref https://github.com/ethereumclassic/ECIPs/pull/285

meowsbits commented:
@soc1c Thanks for review. I need to merge some stuff first, but then will address these things soon!

soc1c commented on params/config_kotti.go:
```diff
@@ -63,13 +63,19 @@ var (
 		EIP152FBlock:  big.NewInt(2058191),
 		EIP1108FBlock: big.NewInt(2058191),
 		EIP1344FBlock: big.NewInt(2058191),
-		EIP1884FBlock: nil,
```
can we keep this line just for the sake of readability? I know it does not solve anything.kk

meowsbits replied:
https://github.com/etclabscore/multi-geth/pull/128/commits/26b180edc4edb435ee8f68c08b118329dcdf87e9

soc1c commented on params/config_classic.go:
```diff
@@ -60,14 +60,18 @@ var (
+		EIP1108FBlock: big.NewInt(10_500_839),
+		EIP1344FBlock: big.NewInt(10_500_839),
+		EIP2028FBlock: big.NewInt(10_500_839),
+		//EIP2200FBlock: big.NewInt(10500839), // RePetersburg (=~ re-1283)
```
can we have enable and disable on the same block number to be super correct and improve readability? I know it wouldn't change anything.

meowsbits replied:
https://github.com/etclabscore/multi-geth/pull/128/commits/26b180edc4edb435ee8f68c08b118329dcdf87e9

meowsbits committed:
miner: fix miner application of DAO state

This causes ETH-supporting chains to incorrectly apply
the DAO state manipulation.

Signed-off-by: meows <b5c6@protonmail.com>
